src/containers/CreateModal.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Modal from "@material-ui/core/Modal";
+import axios from "axios";
+import { useForm } from "react-hook-form";
+
+function rand() {
+  return Math.round(Math.random() * 20) - 10;
+}
+
+function getModalStyle() {
+  const top = 50 + rand();
+  const left = 50 + rand();
+
+  return {
+    top: `${top}%`,
+    left: `${left}%`,
+    transform: `translate(-${top}%, -${left}%)`,
+  };
+}
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    position: "absolute",
+    width: 400,
+    backgroundColor: theme.palette.background.paper,
+    border: "2px solid #000",
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+  },
+}));
+
+export const CreateModal = (props) => {
+  const classes = useStyles();
+
+  // getModalStyle is not a pure function, we roll the style only on the first render
+  const [modalStyle] = React.useState(getModalStyle);
+  const [open, setOpen] = React.useState(false);
+  const { register, handleSubmit, errors } = useForm();
+  const [memberData, setMemberData] = useState(() => {
+    const initialState = props.updateData;
+    return initialState;
+  });
+
+  useEffect(() => {
+    console.log(memberData);
+  });
+
+  // console.log(updateName);
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const onSubmit = async (data) => {
+    console.log(data);
+    // 入力したデータ
+
+    await updateMember(data);
+  };
+  console.log(errors);
+
+  const updateMember = async (data) => {
+    await axios
+      .patch(`http://localhost:4001/api/members/${props.memberId}`, {
+        first_name: memberData,
+      })
+      .then((res) => {
+        console.log(memberData);
+
+        setMemberData({
+          first_name: memberData,
+        });
+      })
+      .catch((data) => {
+        console.log(data);
+      });
+  };
```

Comment:
とりあえず、 react hook formがよし何やってくれるので
入力したデータを自分で useStateで保持しておく必要はないです。

```
const onSubmit = async (data) => {
    console.log(data);
    // 入力したデータ
    
    await updateMember(data);
  };
```
この部分の引数 `data` をconsole.logでみてみたらわかると思うんですけど、ここにデータが入っているので

```
const onSubmit = async (data) => {
    console.log(data);
    // 入力したデータ

    const params = {
      first:name: data.first_name
      // 他も同じように
    }

    await updateMember(params);
  };
```

ってやって

```suggestion
      .patch(`http://localhost:4001/api/members/${props.memberId}`, params)
      .then((res) => {
        console.log(memberData);

        setMemberData({
          first_name: memberData,
        });
      })
      .catch((data) => {
        console.log(data);
      });
  };
```

とすると良さそうですね。
ここはペアプロの時に説明しましょうかねー